ldp/alg/callbacks.py

```diff
@@ -150,24 +150,15 @@ def _make_filename(self, traj_id: str, env: Environment) -> str:
         """Create the filename for the output file."""
         return f"{traj_id}.jsonl"
 
-    async def before_transition(
```

Comment:
We don't need to populate `self.out_files` up front, we can JIT populate within `after_transition`